Patch taken from Alpine: https://git.alpinelinux.org/aports/tree/community/openjdk17/FixNullPtrCast.patch

Subject: Fix cast errors with latest GCC
Upstream: No
Author: Simon Frankenberger <simon-alpine@fraho.eu>

This patch fixes one remaining casting error reported by GCC 12 for aarch64

--- old/src/hotspot/cpu/aarch64/interpreterRT_aarch64.cpp
+++ new/src/hotspot/cpu/aarch64/interpreterRT_aarch64.cpp
@@ -267,7 +267,7 @@
 
   virtual void pass_object() {
     intptr_t* addr = single_slot_addr();
-    intptr_t value = *addr == 0 ? NULL : (intptr_t)addr;
+    intptr_t value = *addr == 0 ? (intptr_t) 0 : (intptr_t)addr;
     if (pass_gpr(value) < 0) {
       pass_stack<>(value);
     }
